dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -318,6 +357,82 @@ public CultureInfo Culture
     }
     #endregion
 
+    #region Internal Filtering
+
+    [Experimental("SKEXP0005")]
+    internal FunctionInvokingContext? OnFunctionInvokingFilter(KernelFunction function, KernelArguments arguments)
+    {
+        FunctionInvokingContext? context = null;
+
+        if (this._functionFilters is { Count: > 0 })
+        {
+            context = new(function, arguments);
+
+            foreach (var filter in this._functionFilters)
+            {
+                filter.OnFunctionInvoking(context);
+            }
+        }
+
+        return context;
+    }
+
+    [Experimental("SKEXP0005")]
+    internal FunctionInvokedContext? OnFunctionInvokedFilter(KernelArguments arguments, FunctionResult result)
+    {
+        FunctionInvokedContext? context = null;
+
+        if (this._functionFilters is { Count: > 0 })
+        {
+            context = new(arguments, result);
+
+            foreach (var filter in this._functionFilters)
```

Comment:
Do we care about the order in which we process the filters?

Reply:
I think by default it should be processed in the same order as filters were initially registered/added. But now there is possibility to access `kernel.FunctionFilters` and `kernel.PromptFilters`, so it's possible to change the order programmatically if needed.